### PR TITLE
AJ-1473: dont show Authorization Domain panel for Azure workspaces

### PR DESCRIPTION
**Pull request by davidangb**
Status 4:45pm Monday Nov 20: The runtime code for this PR, in the first commit, is simple and I have tested it by running locally. In the second commit, I added my attempt at unit tests. The unit tests have some duplicated code that could be cleaned up, but more importantly they are failing with:
```
    TypeError: Cannot read properties of undefined (reading 'makePath')

      68 |     `No handler found for key ${name}. Valid path keys are: ${_.map('name', routeHandlersStore.get())}`
      69 |   );
    > 70 |   return handler.makePath(params);
```
and I need some help resolving that.

-----

## Summary of changes:
Remove the "Authorization Domain" panel for Azure workspaces; leave it for Google workspaces.

### Why
From the Jira ticket:
>Rawls returns group constraint policies as authorization domains. As a result, the authorization domains panel shows on the workspace dashboard for workspaces with group constraint policies.
>
> However, we don’t want to mention “authorization domains” for Azure workspaces. They’ve been replaced by “data access controls” and shown in a different place on the dashboard (AJ-1460).
>
> Thus, we should remove this panel for Azure workspaces. 

### Testing strategy
- [ ] View an Azure workspace with auth domain. The panel should not display.
- [ ] View a Google workspace with auth domain. The panel should display.

### CODE REVIEW

File: src/pages/workspaces/workspace/Dashboard/Dashboard.ts [hunk 1]
```diff
@@ -95,7 +95,7 @@ interface WorkspaceDashboardProps {
   name: string;
 }
 
-interface WorkspaceDashboardComponentProps extends WorkspaceDashboardProps {
+export interface WorkspaceDashboardComponentProps extends WorkspaceDashboardProps {
```
Comment (davidangb) [inline src/pages/workspaces/workspace/Dashboard/Dashboard.ts]: the `export` is only necessary for unit tests; not necessary for runtime

File: src/pages/workspaces/workspace/Dashboard/Dashboard.ts [hunk 2]
```diff
@@ -491,7 +492,7 @@ const WorkspaceDashboardForwardRefRenderFunction = (
   ]);
 };
 
-const WorkspaceDashboard: (props: WorkspaceDashboardProps) => ReactNode = _.flow(
+export const WorkspaceDashboard: (props: WorkspaceDashboardProps) => ReactNode = _.flow(
```
Comment (davidangb) [inline src/pages/workspaces/workspace/Dashboard/Dashboard.ts]: the `export` is only necessary for unit tests; not necessary for runtime

**Comment by cahrens** [comment]
I took a look at this, and I recommend that the very minimal code change goes in without a unit test.

The issue that @davidangb is hitting is that the WorkspaceDashboard component is huge and not written in a testable fashion. It isn't just the RHS components, it's the entire page, and it has a lot of Ajax calls and state management.

Writing a unit tests for it in its current format would require a lot of Ajax mocking unrelated to the code change David is making. In fact, the existing "Dashboard.test.ts" file should be renamed "AzureStorageDetails.test.ts", as that is all it tests, and the `AzureStorageDetails` has been moved out into its own file. I will put in a PR for that update.

I spent a couple of hours trying to extract parts of Dashboard.ts to make whether or not the the Auth Domain section is displayed testable. I kept finding more and more things that needed to be rewritten, and I realized that this task is well beyond the scope of David's particular ticket, as it would require tests to be written for more of the dashboard so that we could be confident that nothing was broken.

**Comment by jladieu** [comment]
@cahrens I'd approve this PR with just the oneliner change, I can park David's test efforts on another branch, and reset this one back to just having the initial commit which we could then both approve if you're ok with that approach.  That way the original PR can land with this conversation attached to it.

**Comment by cahrens** [comment]
Note that I renamed the test in https://github.com/DataBiosphere/terra-ui/pull/4490.

I'm happy to approve the one-line change in either this PR or a new one.

**Comment by jladieu** [comment]
@cahrens I pushed up the reverts and parked the commits with the unit tests on this branch: https://github.com/DataBiosphere/terra-ui/tree/da_AJ-1473_removeAuthDomainAzure_withUnitTests

I'll do some local validation of this change before adding my thumb to the PR, but your 👍 is welcome in the meantime!

**Comment by jladieu** [comment]
Confirmed locally that this logic suppresses the Authorization Domains tab for Azure workspaces.

**Review by cahrens — APPROVED** [review]

**Review by jladieu — APPROVED** [review]

**Comment by jladieu** [comment]
Followup [ticket](https://broadworkbench.atlassian.net/browse/AJ-1483) for unit tests.

**Pull request merged**